workers/generic-worker/tc/factory.go

```diff
@@ -0,0 +1,41 @@
+package tc
+
+import (
+	tcclient "github.com/taskcluster/taskcluster/v29/clients/client-go"
+	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcauth"
+	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcpurgecache"
+	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcqueue"
+	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcsecrets"
+	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcworkermanager"
+)
+
+type ServiceFactory interface {
+	Auth(creds *tcclient.Credentials, rootURL string) Auth
+	Queue(creds *tcclient.Credentials, rootURL string) Queue
+	PurgeCache(creds *tcclient.Credentials, rootURL string) PurgeCache
+	Secrets(creds *tcclient.Credentials, rootURL string) Secrets
+	WorkerManager(creds *tcclient.Credentials, rootURL string) WorkerManager
+}
+
+type ClientFactory struct {
+}
```

Comment:
I like this pattern!  Could this `tc` package go in `internal/` and be used by worker-runner as well?

Also, note that if you like to drop support for talking to worker-manager directly from g-w (since that's now more correctly done by worker-runner), you can save having to mock secrets and worker-manager.  (although worker-runner will need them mocked, but that's already implemented)

Reply:
bump :)

Reply:
Let's move the tc package in a follow up PR - this PR is already getting pretty large and dragging on.